charts/tekton-pipelines/templates/buildpacks/pipelines/_buildpack.yaml

```diff
@@ -2,7 +2,11 @@
 apiVersion: tekton.dev/v1beta1
 kind: Pipeline
 metadata:
-  name: {{ .pipeline.name }}
+  {{- if .buildOnly}}
```

Comment:
Please don't use if/else block for that, you can define `name` variable above, which will be set with `ternary`, for example:

```
{{- $env := ternary .Values.localEnv .Values.prodEnv .Values.isLocal -}}
```